client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelFactory.java

```diff
@@ -282,6 +282,7 @@ private SocketAddress getNextServer(Set<SocketAddress> failedServers, byte[] cac
          if (failedServers != null && failedServers.containsAll(getServers(cacheName))) {
             log.debug("All the servers are marked as failed. Cluster might have completely shut down, try reverting to the initial server list.");
             reset(cacheName);
+            closeChannelPools(failedServers);
```

Comment:
@danberindei I've added this line, which ensures that the pool map is properly cleaned up

Reply:
@tristantarrant `reset` calls `updateServers`, which calls `updateTopologyInfo`, which already does this (that's where the original code of `closeChannelPools` is from). So I'm not sure how this extra call changes anything.

Reply:
I stand corrected @tristantarrant, I didn't realize `failedServers` means something else in `updateTopologyInfo`